cms/templatetags/course_list.py

```diff
@@ -0,0 +1,49 @@
+"""
+Generates a <ul> block for a list of courses. Pass in the list of courses.
+"""
+
+import re
+
+from django import template
+
+from cms.templatetags.feature_img_src import feature_img_src
+
+register = template.Library()
+
+
+def format_course_start_time(start_date):
+    hour = start_date.hour
+    minute = start_date.minute
+    timespec = ""
+
+    if (hour == 23 and minute == 59) or (hour == 0 and minute == 0):
+        timespec = "midnight"
+    else:
+        timespec = re.sub(r"^0", "", start_date.strftime("%I:%M %p"))
+
+    return (
+        f"{start_date.strftime('%b')} {start_date.day}, {start_date.year}, {timespec}"
+    )
+
+
+@register.inclusion_tag("course_list_card.html", name="course_list")
+def course_list(courses):
+    cards = []
+
+    for course in courses:
+        start_descriptor = (
+            f"Starts {format_course_start_time(course.first_unexpired_run.start_date)}"
+            if course.first_unexpired_run.start_date
```

Comment:
I am getting 500 error `AttributeError: 'NoneType' object has no attribute 'start_date'` on this line for programs where one of courses have no open run. The program is linked from homepage and catalog page

Reply:
This is fixed in https://github.com/mitodl/mitxonline/pull/1833/commits/af42117971baf01b19def8c36d855c836c60cbef (I noticed it too, then fixed it, probably at the same time you did)